cognite_toolkit/_cdf_tk/loaders/_resource_loaders/datamodel_loaders.py

```diff
@@ -944,18 +944,14 @@ def dump_resource(self, resource: Node, local: dict[str, Any] | None = None) ->
         # CDF resource does not have properties set, so we need to do a lookup
         local = local or {}
         sources = [ViewId.load(source["source"]) for source in local.get("sources", []) if "source" in source]
+
         if sources:
             try:
-                cdf_resource_with_properties = self.client.data_modeling.instances.retrieve(
-                    nodes=resource.as_id(), sources=sources
-                ).nodes[0]
+                res = self.client.data_modeling.instances.retrieve(nodes=resource.as_id(), sources=sources)
+                dumped = res.nodes[0].as_write().dump() if len(res.nodes) > 0 else resource.as_write().dump()
```

Comment:
I prefer to have this line in the else clause. The except is only catching CogniteAPIError and that cannot be raised from this line.

```python
            try:
                res = self.client.data_modeling.instances.retrieve(nodes=resource.as_id(), sources=sources)
            except CogniteAPIError:
                # View does not exist
                dumped = resource.as_write().dump()
            else:
                dumped = res.nodes[0].as_write().dump() if len(res.nodes) > 0 else resource.as_write().dump()

```
The else claus is only executed if the try runs successfully.